up42/asset.py

```diff
@@ -128,12 +126,28 @@ def update_metadata(
         self._info = response_json
         return self._info
 
-    def _get_download_url(self) -> str:
-        url = f"{self.auth._endpoint()}/v2/assets/{self.asset_id}/download-url"
-        response_json = self.auth._request(request_type="POST", url=url)
+    def _get_download_url(self, stac_asset_id: Optional[str] = None, request_type:str = "POST") -> str:
+        if stac_asset_id is None:
+            url = f"{self.auth._endpoint()}/v2/assets/{self.asset_id}/download-url"
+        else:
+            url = f"{self.auth._endpoint()}/v2/assets/{stac_asset_id}/download-url"
+        response_json = self.auth._request(request_type=request_type, url=url)
         download_url = response_json["url"]
         return download_url
 
+    def get_stac_asset_url(self, stac_asset: pystac.Asset):
+        """
+        Returns the signed URL for the STAC Asset.
```

Comment:
```suggestion
        Returns the pre-signed URL for the STAC Asset.
```
As the URL has a 30 second validity, this would be a pre-signed URL.

Reply:
@saad-up42 maybe I am wrong, but isn't the term "pre-signed URL" very specific to AWS S3 buckets?
If not, what's the difference between signed URL and pre-signed URL? :)

Reply:
@saad-up42 Where did you read about this distinction (limited validity = pre-signed, other = signed)? Because e.g. [here](https://cloud.google.com/storage/docs/access-control/signed-urls) it says "Anyone who knows the _signed_ URL can access the resource until the expiration time for the URL is reached or the key used to sign the URL is rotated.", so seems like Naman is right that it would be considered a signed URL

Reply:
@dustgalactic The [reference](https://stackoverflow.com/questions/20862195/what-is-difference-between-pre-signed-url-and-signed-url) I used also refers to it for AWS S3 buckets, so @seedlit may be right and it might just be a signed URL then.

If that's the case, we can change it back to a signed URL here and in other instances like the API reference.

Reply:
@saad-up42 As for the API reference, don't do it right now, I'll combine it with another task related to the asset repo. 

As for the SDK reference, @seedlit it's your call, we can fix it [in this PR](https://github.com/up42/up42-py/pull/479) we're working on right now or leave it up to you

Reply:
> yeah, let's fix it in the PR you are working on right now :)

@dustgalactic if that's okay I will include the changes in #479 

API changes will need to be made elsewhere.